datadog/resource_datadog_logs_pipeline.go

```diff
@@ -0,0 +1,687 @@
+		"name":                 {Type: schema.TypeString, Required: true},
+		"is_enabled":           {Type: schema.TypeBool, Optional: true},
+		"sources":              {Type: schema.TypeList, Required: true, Elem: &schema.Schema{Type: schema.TypeString}},
+		"source_type":          {Type: schema.TypeString, Required: true},
+		"target":               {Type: schema.TypeString, Required: true},
+		"target_type":          {Type: schema.TypeString, Required: true},
+		"preserve_source":      {Type: schema.TypeBool, Required: true},
+		"override_on_conflict": {Type: schema.TypeBool, Required: true},
+	}
+}
+
+func getCategoryProcessor() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name":       {Type: schema.TypeString, Required: true},
+		"is_enabled": {Type: schema.TypeBool, Optional: true},
+		"target":     {Type: schema.TypeString, Required: true},
+		"category": {Type: schema.TypeList, Required: true, Elem: &schema.Resource{
+			Schema: getCategorySchema(),
+		}},
+	}
+}
+
+func getCategorySchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"filter": {
+			Type:     schema.TypeList,
+			Required: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"query": {Type: schema.TypeString, Required: true},
+				},
+			},
+		},
+		"name": {Type: schema.TypeString, Required: true},
+	}
+}
+
+func getGrokParser() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name":       {Type: schema.TypeString, Required: true},
+		"is_enabled": {Type: schema.TypeBool, Optional: true},
+		"source":     {Type: schema.TypeString, Required: true},
+		"grok": {
+			Type:     schema.TypeList,
+			MaxItems: 1,
+			Required: true,
+			Elem:     &schema.Resource{Schema: getGrokSchema()},
+		},
+	}
+}
+
+func getGrokSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"support_rules": {Type: schema.TypeString, Required: true},
+		"match_rules":   {Type: schema.TypeString, Required: true},
+	}
+}
+
+func getUrlParser() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name":                     {Type: schema.TypeString, Required: true},
+		"is_enabled":               {Type: schema.TypeBool, Optional: true},
+		"sources":                  {Type: schema.TypeList, Required: true, Elem: &schema.Schema{Type: schema.TypeString}},
+		"target":                   {Type: schema.TypeString, Required: true},
+		"normalize_ending_slashes": {Type: schema.TypeBool, Required: true},
+	}
+}
+
+func getUserAgentParser() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name":       {Type: schema.TypeString, Required: true},
+		"is_enabled": {Type: schema.TypeBool, Optional: true},
+		"sources":    {Type: schema.TypeList, Required: true, Elem: &schema.Schema{Type: schema.TypeString}},
+		"target":     {Type: schema.TypeString, Required: true},
+		"is_encoded": {Type: schema.TypeBool, Required: true},
+	}
+}
+
+const (
```

Comment:
I think we tend to define const at the header of the files, from the go library, `buffer` package: https://golang.org/src/bytes/buffer.go?s=14279:14313#L16